ui/src/main/webapp/src/app/services/graph/graph-adjacency.decorator.ts

```diff
@@ -1,23 +1,34 @@
 import {GraphJSONToModelService} from "./graph-json-to-model.service";
 import {Observable} from "rxjs/Observable";
+import {Http, Response} from "@angular/http";
+import {StaticCache} from "./cache";
 
 /**
  * @GraphAdjacency decorator, which handles TypeScript Frames models' property resolving.
  *
  * The underlying object contains graph vertices data or links to graph REST service.
  * This decorator turns these data into model objects, optionally fetching the data first.
+ *
+ * The `Observable`s returned are cached
  */
-export function GraphAdjacency (name: string, direction: string, array: boolean = true): any {
+let emptyArrayObs = Observable.of([]);
+let nullObs = Observable.of(null);
+
+export function GraphAdjacency (name: string, direction: "IN" | "OUT", array: boolean = true): any {
     return function(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
         if (descriptor) {
             descriptor.get = function () {
+                console.log(`* get() v#${this.vertexId} ${name} ${direction}`);
```

Comment:
Is it ok if I just comment them out until this impl settles down? I think we will need them in the near future.